diff --git a/src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs b/src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs
index f95ca8a058..a897486ba9 100644
--- a/src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs
+++ b/src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs
@@ -6,7 +6,7 @@ using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.Authentication;
 
-internal class WebApplicationAuthenticationBuilder : AuthenticationBuilder
+internal sealed class WebApplicationAuthenticationBuilder : AuthenticationBuilder
 {
     public bool IsAuthenticationConfigured { get; private set; }
 
