diff --git a/src/coreclr/pal/src/eventprovider/CMakeLists.txt b/src/coreclr/pal/src/eventprovider/CMakeLists.txt
index aaf420458a6bc..143206b5c34f5 100644
--- a/src/coreclr/pal/src/eventprovider/CMakeLists.txt
+++ b/src/coreclr/pal/src/eventprovider/CMakeLists.txt
@@ -1,6 +1,6 @@
 set(EVENT_MANIFEST ${VM_DIR}/ClrEtwAll.man)
 
-if(CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_FREEBSD)
+if(CLR_CMAKE_HOST_LINUX)
   add_subdirectory(lttngprovider)
 else()
   add_subdirectory(dummyprovider)
