diff --git a/src/redist/targets/GenerateBundledVersions.targets b/src/redist/targets/GenerateBundledVersions.targets
index 2e9f642b8..b25873012 100644
--- a/src/redist/targets/GenerateBundledVersions.targets
+++ b/src/redist/targets/GenerateBundledVersions.targets
@@ -139,6 +139,8 @@
           win-arm64;
           win-x64;
           win-x86;
+	      freebsd-arm64;
+          freebsd-x64;
           " />
 
       <NetCore31RuntimePackRids Include="@(NetCore30RuntimePackRids)"/>
@@ -260,9 +262,11 @@
         linux-x64;
         linux-arm;
         linux-arm64;
+	    freebsd-arm64;
+        freebsd-x64;
         " />
 
-      <Net50Crossgen2SupportedRids Include="linux-musl-x64;linux-x64;win-x64" />
+      <Net50Crossgen2SupportedRids Include="linux-musl-x64;linux-x64;win-x64;freebsd-x64;freebsd-arm64;" />
 
       <Net60Crossgen2SupportedRids Include="
           @(Net50Crossgen2SupportedRids);
@@ -290,6 +294,8 @@
         linux-x64;
         win-arm64;
         win-x64;
+        freebsd-x64;
+        freebsd-arm64;
         " />
 
       <!-- The subset of ILCompiler target RIDs that are officially supported. Should be a subset of
diff --git a/src/redist/targets/GetRuntimeInformation.targets b/src/redist/targets/GetRuntimeInformation.targets
index 6d216ff20..e4494395b 100644
--- a/src/redist/targets/GetRuntimeInformation.targets
+++ b/src/redist/targets/GetRuntimeInformation.targets
@@ -4,7 +4,7 @@
       <HostRid Condition="'$(HostRid)' == '' and '$(MSBuildRuntimeType)' == 'core'">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</HostRid>
       <HostRid Condition="'$(HostRid)' == '' and '$(MSBuildRuntimeType)' != 'core'">win-$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant)</HostRid>
 
-      <IsLinux Condition = " $([MSBuild]::IsOSPlatform('LINUX')) ">True</IsLinux>
+      <IsLinux Condition = " $([MSBuild]::IsOSPlatform('LINUX')) AND ('$(OSName)' == '' OR '$(OSName)' == 'linux') ">True</IsLinux>
       <HostOSName Condition=" '$(HostOSName)' == '' AND $([MSBuild]::IsOSPlatform('WINDOWS')) ">win</HostOSName>
       <HostOSName Condition=" '$(HostOSName)' == '' AND $([MSBuild]::IsOSPlatform('OSX')) ">osx</HostOSName>
       <HostOSName Condition=" '$(HostOSName)' == '' AND $([MSBuild]::IsOSPlatform('FREEBSD')) ">freebsd</HostOSName>
